sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java

```diff
@@ -1,268 +1,303 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//import fr.emse.ci.sparqlext.function.library.FUN_JSONPath;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.Option;
+import com.jayway.jsonpath.spi.json.GsonJsonProvider;
+import com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider;
+import com.jayway.jsonpath.spi.json.JacksonJsonProvider;
+import com.jayway.jsonpath.spi.json.JsonSmartJsonProvider;
+import com.jayway.jsonpath.spi.mapper.GsonMappingProvider;
+
+import fr.mines_stetienne.ci.sparql_generate.SPARQLExt;
+import fr.mines_stetienne.ci.sparql_generate.iterator.IteratorStreamFunctionBase;
+import fr.mines_stetienne.ci.sparql_generate.stream.LookUpRequest;
+import fr.mines_stetienne.ci.sparql_generate.stream.SPARQLExtStreamManager;
+import fr.mines_stetienne.ci.sparql_generate.utils.ContextUtils;
+import fr.mines_stetienne.ci.sparql_generate.utils.LogUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import java.util.*;
+import java.util.function.Consumer;
+import org.apache.commons.io.IOUtils;
+import org.apache.jena.atlas.web.TypedInputStream;
+import org.apache.jena.graph.Node;
+import org.apache.jena.riot.SysRIOT;
+
+import org.apache.jena.sparql.expr.ExprEvalException;
+import org.apache.jena.sparql.expr.ExprList;
+import org.apache.jena.sparql.expr.NodeValue;
+import org.apache.jena.sparql.expr.nodevalue.NodeValueBoolean;
+import org.apache.jena.sparql.expr.nodevalue.NodeValueInteger;
+import org.jsfr.json.JsonPathListener;
+import org.jsfr.json.JsonSurfer;
+import org.jsfr.json.JsonSurferGson;
+import org.jsfr.json.ParsingContext;
+import org.jsfr.json.SurfingConfiguration;
+import org.jsfr.json.compiler.JsonPathCompiler;
+import org.jsfr.json.exception.JsonSurfingException;
+import org.jsfr.json.path.JsonPath;
+import org.jsfr.json.provider.JsonProvider;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+/**
+ * Iterator function
+ * <a href="http://w3id.org/sparql-generate/iter/JSONSurfer">iter:JSONSurfer</a>
+ * extracts a list of sub-JSON documents of a JSON document, according to a
+ * forward-only JSONPath expression. See https://github.com/jsurfer/JsonSurfer
+ * for the JSONPath syntax specification.
+ *
+ * <p>
+ * See <a href=
+ * "https://w3id.org/sparql-generate/playground.html#ex=example/generate/Example-JSONSurfer">Live

```

Comment:
Please create an example named this way

Reply:
An example is added, and other comments were taken into account.